app/controllers/saml_idp_controller.rb

```diff
@@ -56,7 +56,7 @@ def render_template_for(message, action_url, type)
   end
 
   def delete_branded_experience
-    session.delete(:sp) if session[:sp]
-    session.delete(:user_return_to) if session[:user_return_to]
+    session.delete(:sp)
+    session.delete('user_return_to')
```

Comment:
Per my commit message, this change was because Devise uses a string and I wanted to make sure to use the same thing Devise uses to avoid surprises. https://github.com/plataformatec/devise/blob/88724e10adaf9ffd1d8dbfbaadda2b9d40de756a/lib/devise/controllers/store_location.rb#L54